### DISCUSSION TITLE
resize/downsample is significantly blurrier than skimage

### DISCUSSION THREAD

**Discussion by scott-vsi** [kornia/kornia#1596] [Potential Issue]
When downsampling an image (512,512) to (224,224), I have noticed that kornia produces a much blurrier image than skimage. 

It looks like kornia uses a sigma [see [1](https://github.com/kornia/kornia/blob/62f114e7883292ae6bb38874cfaca2d98c13a160/kornia/geometry/transform/affwarp.py#L577) [2](https://github.com/kornia/kornia/blob/62f114e7883292ae6bb38874cfaca2d98c13a160/kornia/geometry/transform/affwarp.py#L584)] computed by

```python
factors = (h / size[0], w / size[1])
sigmas = (max(factors[0], 1.0), max(factors[1], 1.0))
```

for the gaussian filtering when down-scaling (where `size` is the output size), whereas skimage uses `(s - 1) / 2` where `s` is the down-scaling factor [see [3](https://github.com/scikit-image/scikit-image/blob/70fa904eee9ef370c824427798302551df57afa1/skimage/transform/_warps.py#L122) and specifically [4](https://github.com/scikit-image/scikit-image/blob/70fa904eee9ef370c824427798302551df57afa1/skimage/transform/_warps.py#L171)].

For example,

```python
import torch, kornia, skimage.transform, numpy as np

image = skimage.data.astronaut()
kornia_out = kornia.geometry.transform.resize(torch.tensor(image/255.0).permute(2,0,1),
        size=(224, 224), interpolation='bilinear', antialias=True)
kornia_out = kornia_out.numpy().transpose(1,2,0)
# vs
skimage_out = skimage.transform.resize(image, (224, 224), order=1, anti_aliasing=True) # bilinear
```

In this case, skimage uses a sigma of 0.64 whereas kornia uses 2.28. Is it expected that these two should be so different?

**Comment by edgarriba** [comment]
@scott-vsi thanks to rep. @ducha-aiki was the one implemented that functionality, he might know the answer :)

**Reply by edgarriba** [reply]
In fact, I like their api where you can pass the sigma. Probably we should try to match them ?

**Comment by scott-vsi** [comment]
Actually, the behavior should probably be considered a bug. Even resizing from 512 to 511 results in a heavy blur (sigma=1.002) being applied. I'm sure a DSP expert could provide a principled formula, but at the very least I would copy what skimage does (sigma=0.00098).

**Reply by ducha-aiki** [reply]
I have implemented anti-aliasing mode having in mind downscaling 2x and more - then it makes sense. If your use-case is very small downscale, it is probably better to turn it off.
 But you are right, it should handle smaller ratios automatically

**Reply by ducha-aiki** [reply]
https://github.com/kornia/kornia/pull/1612 here is the fix
